Plugins~/Src/MeshSyncClientBlender/python/3.1.0/unity_mesh_sync.py

```diff
@@ -143,6 +149,7 @@ def invoke(self, context, event):
             if not MESHSYNC_OT_AutoSync._registered:
                 context.window_manager.modal_handler_add(self)
                 MESHSYNC_OT_AutoSync._registered = True
+                bpy.app.handlers.load_post.append(MESHSYNC_OT_AutoSync.load_handler)
```

Comment:
_registered is reset on file load? the root cause is not clear to me, the comments above seem to acknowledge the issue.

Reply:
Blender clears the modal handler callback when a file is loaded, with this change we know when a file is loaded and the handler is cleared and can register it again. The API doesn't give us a way to remove handlers or check whether or not they are registered so this is the only way we can deal with it.

Reply:
That makes sense , thank you!

Reply:
So, I tried the sample modal operator in this [link](https://docs.blender.org/api/current/bpy.types.Operator.html) (under "Modal Execution"), and it seems that the modal handler is unregistered when we return `FINISHED` or `CANCELLED`.
Can you please confirm this ? And if so, maybe we just need to return FINISHED when opening a new blend file, rather than using additional _registered check ?

Reply:
This operator always returns {'PASS_THROUGH'} from its 'modal' method, so it's never unregistered.

Reply:
If we make it return FINISHED, the auto-sync would stop working. It needs to return PASS_THROUGH so that it gets called until autosync is disabled. 
We only want to register the handler when auto-sync is on, if I add a handler to load_post, that would also trigger when a file is reloaded but I just want to set the flag to know whether it's registered or not, loading a file also resets modal operators, so we would not know whether or not it shouuld be running and users probably don't want to keep this running when a new file is loaded anyway.

Reply:
Ah, so you mean remove the _registered flag altogether and just call context.window_manager.modal_handler_add(self) when a file is loaded and a new operator is created?
I've tried that but we have no reference to the operator in the load callback, it's a class method. Also, when the operator is created, we cannot register in its init method because we can't access the context. Operators are supposed to do stuff in their execute method.

Reply:
No, I meant that we could keep using `load_handler` but only set it up once during the entire Blender session.

```
bpy.app.handlers.load_post.append(MESHSYNC_OT_AutoSync.load_handler)
```

I guess there are a couple of places which would be good candidates to put this code:
1. [near the end of the py script](https://github.com/Unity-Technologies/MeshSyncDCCPlugins/blob/31c5f6599885f6a6fa07c9b740f1306eb385f8cc/Plugins~/Src/MeshSyncClientBlender/python/3.1.0/unity_mesh_sync.py#L353)
2. in [register() ](https://github.com/Unity-Technologies/MeshSyncDCCPlugins/blob/31c5f6599885f6a6fa07c9b740f1306eb385f8cc/Plugins~/Src/MeshSyncClientBlender/python/3.1.0/unity_mesh_sync.py#L317). If we want to ensure the cleanup, we can also remove `load_handler` in [unregister()](https://github.com/Unity-Technologies/MeshSyncDCCPlugins/blob/31c5f6599885f6a6fa07c9b740f1306eb385f8cc/Plugins~/Src/MeshSyncClientBlender/python/3.1.0/unity_mesh_sync.py#L322) as well, although I don't think this is necessary.

And a friendly reminder to please replicate the updates to other py files if you make these changes.

Reply:
Ok, I've changed it to do that instead.